tests/ecosystem/upgrade/conftest.py

```diff
@@ -0,0 +1,309 @@
+import logging
+
+from ocs_ci.ocs import constants, resources
+import pytest
+import threading
+
+log = logging.getLogger(__name__)
+
+
+def create_pods(
+    interface,
+    pvc_factory,
+    pod_factory,
+    storageclass,
+    count,
+    access_mode,
+    volume_mode=None
+):
+    """
+    Create pods for upgrade testing. pvc_factory and pod_factory have to be
```

Comment:
Not clear why the fixture has to be specific to upgrade testing, IMHO the choice should be left to users to do what they want to do in upgrade as per testcases, eg

```python
      with parallel() as p:
                 do_upgrade()
                 create_pods()
                 run_io()
                 destroy_pods()
      post_verification()
```

Reply:
This is a function that helps to prepare pods for my test case. You mean pre_upgrade_block_pods and pre_upgrade_filesystem_pods fixtures? These fixtures should be only upgrade specific as there are test cases with markers @order_pre_upgrade and @order_post_upgrade that use them and control what they do during upgrade (eg. test_start_pre_upgrade_pod_io).